lib/ruby_lsp/requests/code_action_resolve.rb

```diff
@@ -0,0 +1,105 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Requests
+    # ![Code action resolve demo](../../misc/code_action_resolve.gif)
+    #
+    # The [code action resolve](https://microsoft.github.io/language-server-protocol/specification#codeAction_resolve)
+    # request is used to to resolve the edit field for a given code action, if it is not already provided in the
+    # textDocument/codeAction response. We can use it for scenarios that require more computation such as refactoring.
+    #
+    # Currently one use case is implemented: Extract to variable.
+    #
+    # # Example
+    #
+    # ```ruby
+    # # Before:
+    # 1 + 1 # Select the text and use Refactor: Extract variable
+    #
+    # # After:
+    # new_variable = 1 + 1
+    # new_variable
+    #
+    # ```
+    #
+    class CodeActionResolve < BaseRequest
+      extend T::Sig
+      NEW_VARIABLE_NAME = "new_variable"
+
+      class CodeActionError < StandardError; end
+
+      class Error < ::T::Enum
+        enums do
+          EmptySelection = new
+          UnsupportedAction = new
+        end
+      end
+
+      sig { params(document: Document, code_action: T::Hash[Symbol, T.untyped]).void }
+      def initialize(document, code_action)
+        super(document)
+
+        @code_action = code_action
+      end
+
+      sig { override.returns(T.any(Interface::CodeAction, Error)) }
+      def run
+        return Error::UnsupportedAction unless @code_action[:kind] == Constant::CodeActionKind::REFACTOR_EXTRACT
```

Comment:
The server only receives resolve requests for the code actions we return ourselves. Is there a scenario in which we would receive an unsupported resolve?